src/v2_0/rw/elog.cpp

```diff
@@ -84,54 +84,82 @@ namespace abb ::rws ::v2_0 ::rw ::elog
         return elogDomains;
     }
 
-    std::vector<std::pair<int, std::string>> getElogDomains(RWSClient &client, std::string lang)
+    std::vector<std::pair<int, std::string>> getElogDomains(RWSClient& client, std::string lang)
     {
         std::stringstream uri;
         uri << "/rw/elog?lang=" << lang << "&resource=count";
         POCOResult poco_result = client.httpGet(uri.str());
-        RWSResult const &rws_result = parseXml(poco_result.content());
+        RWSResult const& rws_result = parseXml(poco_result.content());
         return parseDomains(rws_result);
     }
 
-    ElogMessage parseElogMessageXml(std::string const xml_string)
+    std::string concatTextFromNodes(std::vector<Poco::XML::Node*> nodes, std::string separator = " ")
+    {
+        std::stringstream ss;
+        bool separator_flag = false;
+        for (auto node : nodes)
+        {
+            if (separator_flag)
+                ss << separator;
+            separator_flag = true;
+            ss << node->innerText();
+        }
+        return ss.str();
+    }
+
+    ElogMessage parseElogNode(Poco::XML::Node* node)
     {
-        Poco::AutoPtr<Poco::XML::Document> rws_result = parseXml(xml_string);
         ElogMessage message;
-        message.messageType = makeElogMessageType(std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "msgtype"))));
-        message.code = std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "code")));
-        message.sourceName;
-        std::istringstream ss(xmlFindTextContent(rws_result, XMLAttribute("class", "tstamp")));
+
+        std::string title = xmlNodeGetAttributeValue(node, "title");
+        int sep_pos = title.find_last_of("/");
+        message.domain = std::stoi(title.substr(9, sep_pos - 9));
+        message.sequenceNumber = std::stoi(title.substr(sep_pos + 1));
```

Comment:
Maybe we should throw some paring error if we fail to get the parts of the string, since those funtions can throw multiple excetions: bad_alloc, out_of_range, invalid_argument according to:
https://cplusplus.com/reference/string/string/substr/
https://en.cppreference.com/w/cpp/string/basic_string/stol